swap/swap_test.go

```diff
@@ -736,6 +738,9 @@ func TestDebtCheques(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	testBackend.cashDone = make(chan struct{})
+	defer close(testBackend.cashDone)
```

Comment:
i find this kind of strange to put this here. I've noticed this is not a new thing but I don't see any reason why the test case should close this channel.
Also, another note on the usage of this channel - since it should only signal once, you can close it directly in the method that polls the cashing of the cheque. There's no need to put an empty struct in the channel.

Reply:
Do you mean it's strange because it should be closed elsewhere or because it doesn't need closing at all? I'm not sure what the motivation behind the close was, as far as I understand go, the gc would take care of this anyway.

There are tests (e.g. the simulation tests) where cashing happens multiple times, hence the use of the `struct{}{}` instead of close.

Reply:
I got rid of the `make` in the tests and removed the `close`.

Reply:
> Do you mean it's strange because it should be closed elsewhere or because it doesn't need closing at all?

I mean it in the sense that a channel's reader should not close it. This is a well defined anti-pattern in `go`